runtime/strategies/match-recipe-by-verb.js

```diff
@@ -71,26 +71,27 @@ export default class MatchRecipeByVerb extends Strategy {
 
             particleForReplacing.remove();
 
-
             for (let consumeSlot in slotConstraints) {
-              if (slotConstraints[consumeSlot].targetSlot || Object.values(slotConstraints[consumeSlot].providedSlots).filter(a => a != null).length > 0) {
+              if (slotConstraints[consumeSlot].tags.length > 0 || slotConstraints[consumeSlot].targetSlot || Object.values(slotConstraints[consumeSlot].providedSlots).filter(a => a != null).length > 0) {
```

Comment:
this is supposed to be checking whether there's a concrete slot that's already been mapped by the containing recipe. I don't think you want to add the tags check here?